framework/src/base/MooseError.C

```diff
@@ -0,0 +1,26 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "MooseError.h"
+
+namespace Moose {
+
+void MPIAbortWrapper()
+{
+#if defined(LIBMESH_HAVE_MPI)
+  MPI_Abort(libMesh::GLOBAL_COMM_WORLD, 1);
+#endif
```

Comment:
No, that is not the difference. Both exit and std::abort seem to be _magic_ (in C++11 they are tagged `noreturn` but prior to that I assume the compilers use a heuristic). They both disable the warning if they are in in the body of teh function, but not when they are wrapped in another function call.